### DISCUSSION TITLE
Issue with transitions / animations

### DISCUSSION THREAD

**Discussion by ganeshkrishnasharma** [chakra-ui/chakra-ui#9156] [General]
I am facing an issue with transitions in the following case

I have a web application built using Chakra UI V2. Recently i developed a web SDK using chakra UI V3. When i use this SDK in my web application , the transitions are not working. ( Transition in Chakra UI v3 spinner component is not working & my css properties like `transition: "stroke-dashoffset 0.1s ease"` not working when the SDK is used in an application developed using Chakra UI V2 )

@segunadebayo please kindly help me on this with your suggestions.

**Comment by ganeshkrishnasharma** [comment]
Any help on this is much appreciated.

**Reply by segunadebayo** [reply]
Sharing a reproduction is the best way to get help. Could you share one?

**Reply by ganeshkrishnasharma** [reply]
Hi @segunadebayo unfortunately i have some issues related to sharing the code. I will share you more details here. 

A small correction - The transition is working, but the animation properties are not being applied correctly. Please see below, the animation color is not applied.

Screenshot of the issue
![Screenshot 2024-11-15 at 9 57 09 AM](https://github.com/user-attachments/assets/1fc7be31-3d54-4669-b6ba-87f6dcae4d47)
![Screenshot 2024-11-15 at 9 55 14 AM](https://github.com/user-attachments/assets/5c5681bc-c519-457d-9a2b-e851219600fc)


If i am using v3 only, the CSS for spinner is getting added like this
```
.css-rvexgr {
    color: var(--vd-colors-loading-spinner) !important;
    width: 5rem;
    height: 5rem;
    border-width: 0.5rem;
    -webkit-animation: var(--vd-animations-spin);
    animation: var(--vd-animations-spin);
    -webkit-animation-duration: var(--vd-durations-slow);
    animation-duration: var(--vd-durations-slow);
}
```

When i use v3 for SDK and add it as a script to a web app using v2, this is how CSS is getting added.
```
.css-rvexgr {
    width: 5rem;
    height: 5rem;
    animation-timing-function: ;
    animation-delay: ;
    animation-iteration-count: ;
    animation-direction: ;
    animation-fill-mode: ;
    animation-play-state: ;
    animation-name: ;
    animation-timeline: ;
    animation-range-start: ;
    animation-range-end: ;
    animation-duration: var(--vd-durations-slow);
    color: var(--vd-colors-loading-spinner) !important;
    border-width: 0.5rem;
}
```

Please check if you can provide some suggestions looking at this. Thanks in advance!

**Reply by ganeshkrishnasharma** [reply]
I think this is the issue

```
*, *::before, ::after {
    border-color: var(--chakra-colors-chakra-border-color);
}

<style>
:where(*, *::before, *::after) {
    border-width: 0;
    border-style: solid;
    box-sizing: border-box;
    word-wrap: break-word;
}
```

this style is getting added globally & `var(--chakra-colors-chakra-border-color)` this is unavailable in v3

**Comment by ganeshkrishnasharma** [comment]
```
*, *::before, ::after {
    border-color: var(--chakra-colors-chakra-border-color);
}
```

Not sure where this is getting added in Chakra V2. I tried all the ways in my Chakra Ui v3 SDK, like overriding this using globalCss in theme, adding higher specificity, nothing seems to work :(

**Reply by ganeshkrishnasharma** [reply]
@segunadebayo , any suggestions on how i can over ride this please?

**Reply by ganeshkrishnasharma** [reply]
Hi @segunadebayo , please recommend

```
*, *::before, ::after {
    border-color: var(--chakra-colors-chakra-border-color);
}
```

this style is getting applied from chakra ui v2 , and i am not able to override this either in base application, or in my sdk code. Could you please help? Thanks in advance

**Comment by ganeshkrishnasharma** [comment]
https://github.com/chakra-ui/chakra-ui/issues/9256 

this answers this discussion.
